src/lexer/mod.rs

```diff
@@ -171,7 +187,13 @@ impl Lexer {
     }
     fn new_token(&mut self, token_type: TkType, value: String) -> Token {
         Token(
-            Location::new(self.file_name.clone(), self.line, self.pos),
+            Location::new(
+                self.file_name.clone(),
+                self.line,
+                self.pos,
+                self.start as u32,
+                self.offset as u32,
```

Comment:
This is, unfortunately, that we cannot convert `lexer.start` and `lexer.end` to `u32`, so we have to take a dangerous about type conversion.